lib/workers/repository/updates/branch-name.spec.ts

```diff
@@ -293,5 +293,35 @@ describe('workers/repository/updates/branch-name', () => {
         expect(fixture.upgrade.branchName).toEqual(fixture.expectedBranchName);
       });
     });
+
+    it('strict branch slugify enabled', () => {
+      const upgrade: RenovateConfig = {
+        strictBranchSlugify: true,
+        groupName: '[some] group name.#$%version',
+        group: {
+          branchName: '{{groupSlug}}-{{branchTopic}}',
+          branchTopic: 'grouptopic',
+        },
+      };
+      generateBranchName(upgrade);
+      expect(upgrade.branchName).toBe(
+        'some-group-namedollarpercentversion-grouptopic'
```

Comment:
This not what I'd been expecting actually. Had thought they'd be replaced with `-`

Reply:
That makes two of us.  I was expecting the same before diving deeper into how the slugify strict mode behaves.

My alternate suggestion would be to have `strictBranchSlugify` control an additional replace regex within [cleanBranchName](https://github.com/renovatebot/renovate/blob/c3e5e07344abc122be3427cc8ef11e3c6c7c6b81/lib/workers/repository/updates/branch-name.ts#L22)

Reply:
Another suggestion, but a bit more out there, might be to use the [slug package](https://www.npmjs.com/package/slug) instead of slugify.  The differences our listed [here](https://github.com/Trott/slug#differences-between-slug-and-slugify-packages)

It doesn't help with the `.` -> `-`, but does ensure any special characters are removed, leading and trailing `-` are auto-trimmed AND no default character substitution takes place (eg: special characters, emojis, etc)

All that said, I'm still leaning towards using `strictBranchSlugify` to control an additional regex within `cleanBranchNames`

Reply:
Change approach and updated the PR.  Add new regex which will behave closer to as expected.  

slugify is still doing a handful of symbol character substitution, which is existing behavior